### PR TITLE
Check update handler

### PR DESCRIPTION
**Pull request by danewalton**
Check the update handler to make sure it's a supported handler.

### CODE REVIEW

File: demos/sample_azure_iot_adu/sample_azure_iot_pnp_simulated_data.c [hunk 1]
```diff
@@ -370,6 +397,11 @@ static bool prvDoesInstalledCriteriaMatchCurrentVersion( const AzureIoTADUUpdate
  */
 static AzureIoTADURequestDecision_t prvUserDecideShouldStartUpdate( AzureIoTADUUpdateRequest_t * pxAduUpdateRequest )
 {
+    if( !prvIsHandlerSupported( pxAduUpdateRequest ) )
+    {
+        return eAzureIoTADURequestDecisionReject;
```
Comment (vaavva) [inline demos/sample_azure_iot_adu/sample_azure_iot_pnp_simulated_data.c:402]: Should we add a Log here like the other rejects to explain what went wrong?

**Review by vaavva — APPROVED** [review]

File: demos/sample_azure_iot_adu/sample_azure_iot_pnp_simulated_data.c [hunk 2]
```diff
@@ -23,6 +23,8 @@
 
 /*-----------------------------------------------------------*/
 
+#define sampleazureiotUPDATE_HANDLER                      "microsoft/swupdate:1"
```
Comment (ericwol-msft) [inline demos/sample_azure_iot_adu/sample_azure_iot_pnp_simulated_data.c:26]:
```suggestion
#define sampleazureiotUPDATE_HANDLER                      "contoso/swupdate:1"
```

File: demos/sample_azure_iot_adu/sample_azure_iot_pnp_simulated_data.c [hunk 3]
```diff
@@ -356,6 +358,31 @@ static bool prvDoesInstalledCriteriaMatchCurrentVersion( const AzureIoTADUUpdate
...
 /*-----------------------------------------------------------*/
 
+/**
+ * @brief Verifies that the handler is supported
+ *
+ * @param pxAduUpdateRequest Parsed update request, with the ADU update manifest.
+ * @return true If the handler for the update step matches the supported handler.
+ * @return false If the handler for the update step does not match the supported handler.
+ */
+static bool prvIsHandlerSupported( const AzureIoTADUUpdateRequest_t * pxAduUpdateRequest )
```
Comment (ericwol-msft) [inline demos/sample_azure_iot_adu/sample_azure_iot_pnp_simulated_data.c:368]:
```suggestion
static bool prvIsADUHandlerSupported( const AzureIoTADUUpdateRequest_t * pxAduUpdateRequest )
```

Comment (ericwol-msft) [reply demos/sample_azure_iot_adu/sample_azure_iot_pnp_simulated_data.c:368]: or prvIsUdateHandlerSupported

**Review by ericwol-msft — APPROVED** [review]

**Review by wduraes — APPROVED** [review]

File: demos/sample_azure_iot_adu/sample_azure_iot_pnp_simulated_data.c (hunk 2 above)
Comment (danewalton) [reply demos/sample_azure_iot_adu/sample_azure_iot_pnp_simulated_data.c:26]: The handler name as is an ADU defined term

**Pull request merged**